frontend/src/components/common/wrappers/AuthorizationWrapper.js

```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import { PropTypes } from "prop-types";
+import { connect } from "react-redux";
+import { getUserAccessData } from "@/selectors/authenticationSelectors";
+import { USER_ROLES } from "@/constants/environment";
+
+/**
+ * @constant AuthorizationWrapper conditionally renders react children depending
+ * if the role passed in matches the user permissions.
+ * 
+ * NOTE: If the childComponent is an ant design child component**
+ * IE. Menu.Item,
+ * <Menu>
+ *    <Menu.Item />
+ * </Menu>
+ *  The implicit flow of passing props down to children could be interrupted.
+ * Ie, Menu passes onItemHover down to Menu.Item, with AuthorizationWrapper wrapping Menu.Item, it no longer has access
+ * to onItemHover, as a solution, a <div className="custom-menu-item" /> will be used in place of Menu.Item and will inherit all CSS from 
+ * Menu.Item.
+ * NavBar.js use case example: 
+ * <Menu>
+    <AuthorizationWrapper permission="role_admin">
+      <div className="custom-menu-item">
+        <Link to={router.ADMIN_DASHBOARD.route}>
+          Admin
+        </Link>
+      </div>
+    </AuthorizationWrapper>
+  </Menu>
+ * 
+ *
+ */
+
+const propTypes = {
+  permission: PropTypes.string.isRequired,
+  children: PropTypes.element.isRequired,
+};
+
+export const AuthorizationWrapper = (props) =>
+  props.userRoles.includes(USER_ROLES[props.permission]) && <div>{...props.children}</div>;
```

Comment:
As we have constants defined for these, would it be better to use the constants in each component as well? Pass USER_ROLES.role_create as the permission prop instead of finding it here? 

Alternatively, we could use the role name directly in each view "mds-mine-admin". This approach seems like we still using magic strings in our components, but adding another set of magic strings we need to know. I don't really have strong feelings here, but thoughts on this?

Reply:
Yeah I was thinking about passing the permissions around as constants instead of strings.

The reason I opted against it is because our permissions are so simple at this point, I was considering moving them into constant files when we make them more complex. 

I also don't have strong feelings, and will do which ever you think is better at this stage.

Reply:
I'm happy with roles as variables 💯